cmd/goose/main.go

```diff
@@ -64,6 +64,10 @@ func main() {
 		driver = "postgres"
 	}
 
+	if driver == "tidb" {
+		driver = "mysql"
+	}
```

Comment:
shouldn't this be in the switch driver statement instead?

Reply:
Hmm, I just assumed to put it there, since redshift does the same thing just above. TiDB uses MySQL driver, but the added TiDB dialect for creating the goose_db_version table.

Reply:
yeah, both `tidb` and `redshift` should call `goose.SetDialect(driver);`, right?

if I'm not wrong, then it should make sense to move them to the new `case` statement, along with "postgres"

Reply:
You mean like this?

```go
switch driver {
	case "postgres", "mysql", "sqlite3", "redshift":
		if err := goose.SetDialect(driver); err != nil {
			log.Fatal(err)
		}
	case "tidb":
		if err := goose.SetDialect(driver); err != nil {
			log.Fatal(err)
		}
		driver = "mysql"
	default:
		log.Fatalf("%q driver not supported\n", driver)
	}
```

Reply:
```go
switch driver {
case "redshift":
	driver = "postgres"
case  "tibd":
	driver = "mysql"
}
if err := goose.SetDialect(driver); err != nil {
	log.Fatal(err)
}
```

Reply:
I'd expect `goose.SetDialect()` to error out on unsupported driver.

Reply:
No, that wouldn't work. driver needs to be tidb when setting dialect in goose so that it uses the method with correct receiver type `type TiDBDialect struct{}`, this way goose create a version table with the correct syntax (No serial alias) for TiDB. Then we switch driver to MySQL so that it actually uses the official MySQL driver `github.com/go-sql-driver/mysql`

Reply:
I see, how about 

```go
if err := goose.SetDialect(driver); err != nil {
	log.Fatal(err)
}

switch driver {
case "redshift":
	driver = "postgres"
case  "tibd":
	driver = "mysql"
}
```
then?

Reply:
That would work. Changing it.